src/main/java/one/util/streamex/EntryStream.java

```diff
@@ -196,6 +177,32 @@
                 stream().flatMap(e -> withValue(mapper.apply(e.getKey(), e.getValue()), e.getValue())), context);
     }
 
+    /**
+     * Returns an {@code EntryStream} consisting of the entries whose values are
+     * results of replacing source values with the contents of a mapped stream
+     * produced by applying the provided mapping function and converting the
+     * resulting optional to a stream.
+     *
+     * Keys are left intact.
+     *
+     * <p>
+     * This is an <a href="package-summary.html#StreamOps">intermediate
+     * operation</a>.
+     *
+     * @param <KK> The type of new keys
+     * @param mapper a <a
+     *        href="package-summary.html#NonInterference">non-interfering </a>,
+     *        <a href="package-summary.html#Statelessness">stateless</a>
+     *        function to apply to each key and value which produces an optional key
+     * @return the new stream
+     * @since 0.6.8
+     */
+    public <KK> EntryStream<KK, V> flatOptionToKey(

```

Comment:
This name is definitely unsatisfactory. `flatOptionalKeys` or `mapKeysPartial`? The same with other names in `EntryStream`.